.github/workflows/release.yaml

```diff
@@ -239,3 +238,46 @@ jobs:
           sbom: 'result.json'
           signing-key: ${{ secrets.COSIGN_PRIVATE_KEY }}
           signing-password: ${{ secrets.COSIGN_PASSWORD }}
+  pre-release:
+    if: ${{ contains(github.ref, '-rc.') }}
+    name: Create pre-release
+    needs: [prepare]
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - name: Setup Golang
+        uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
+        with:
+          go-version-file: "${{ github.workspace }}/go.mod"
+      - name: Download dependencies
+        id: depdownload
+        run: |
+          hack/build/ci/install-cgo-dependencies.sh
+      - name: Generate release notes
+        id: generate_notes
+        run: |
+          hack/build/ci/generate-release-notes.sh
+      - name: Generate manifests
+        run: |
+          export IMAGE=docker.io/${{ secrets[matrix.repository] }}
```

Comment:
Have you tested this? you have not specified a matrix within your step, therefore I think this will not work as matrix.repositry is not defined, also you could replace the first part `docker.io` with the properly defined registry. You can look at line 53 how its done in that step